### ISSUE TITLE
Monitor training process with batch hard

### ISSUE THREAD

**Issue by FSet89** [omoindrot/tensorflow-triplet-loss#28]
When using batch hard, is it correct to say that the hardest negative distance should increase and the hardest positive distance should decrease during training? Is it a good method to monitor the training process?

**Comment by paweller** [comment]
Yes, it is true that the inter-class (or negative) distances should increase while the intra-class (or positive) distance should decrease during training. The triplet loss aims to do exactly that.

Regarding the monitoring process you can use the mean inter-class or intra-class distance. However, as those can differ a lot from project to project, you might want to rather use other metrics. These could be a triplet error rate (`number_of(d(A,P)-d(A,N) > 0) / number_total_triplets`) or the [AUC-ROC](https://www.analyticsvidhya.com/blog/2020/06/auc-roc-curve-machine-learning/) metric or simply the L2 distance.
